### PR TITLE
Additional tests

### PR DESCRIPTION
**Pull request by olivdb**

### CODE REVIEW

File: test/wormhole.test.ts [hunk 1]
```diff
@@ -271,12 +259,70 @@ describe('Wormhole', () => {
...
+      expect(await l2WormholeBridge.batchedDaiToFlush(mainnetDomain)).to.be.eq(amt)
+      expect(await mainnetSdk.dai.balanceOf(l1Escrow.address)).to.be.eq(amt)
+
+      // Pay back (not yet incurred) debt. Usually relaying this message would take 7 days
+      await relayMessagesToL1(l2WormholeBridge.connect(l2User).flush(mainnetDomain))
+
+      expect(await l2WormholeBridge.batchedDaiToFlush(mainnetDomain)).to.be.eq(0)
+      expect(await mainnetSdk.dai.balanceOf(l1Escrow.address)).to.be.eq(0)
+      expect(await mainnetSdk.dai.balanceOf(router.address)).to.be.eq(0)
+      expect(await mainnetSdk.dai.balanceOf(join.address)).to.be.eq(0)
```
Comment (krzkaczor) [inline test/wormhole.test.ts:274]: can we add here (and in the next testcase) an assert against debt ceiling here? dssWormholeJoin.debt and internal accounting in vat (`dai`?)

File: test/spell.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  )
+  const badDebtPushSpell = await deployUsingFactory(opts.l1Signer, BadDebtPushSpellFactory, [
+    opts.wormholeJoinAddress,
+    opts.sdk.vat.address,
+    opts.sdk.dai_join.address,
+    opts.sdk.vow.address,
+    opts.sourceDomain,
+    opts.badDebt,
+  ])
+  const castBadDebtPushSpell = opts.sdk.pause_proxy
```
Comment (krzkaczor) [inline test/spell.ts:32]: can we make this a function instead? I would expect that deploySpell just deploys spell not deploys and executes it.

**Review by krzkaczor — CHANGES_REQUESTED** [review]

File: scripts/copy-artifacts.js [hunk 3]
```diff
@@ -48,9 +49,13 @@ copyDappToolsArtifact(dappToolsArtifacts, 'WormholeJoin', output)
...
-copyHardhatArtifact(join(hardhatArtifacts, 'l2/L2DAIWormholeBridge.sol/L2DAIWormholeBridge.json'), output)
+copyHardhatArtifact(join(hardhatBridgeArtifacts, 'l1/L1DAITokenBridge.sol/L1DAITokenBridge.json'), output)
+copyHardhatArtifact(join(hardhatBridgeArtifacts, 'l2/dai.sol/Dai.json'), output)
+copyHardhatArtifact(join(hardhatBridgeArtifacts, 'l2/L2DAITokenBridge.sol/L2DAITokenBridge.json'), output)
+copyHardhatArtifact(join(hardhatBridgeArtifacts, 'l1/L1DAIWormholeBridge.sol/L1DAIWormholeBridge.json'), output)
+copyHardhatArtifact(join(hardhatBridgeArtifacts, 'l1/L1Escrow.sol/L1Escrow.json'), output)
+copyHardhatArtifact(join(hardhatBridgeArtifacts, 'l2/L2DAIWormholeBridge.sol/L2DAIWormholeBridge.json'), output)
+copyHardhatArtifact(join(hardhatTestArtifacts, 'TestBadDebtPushSpell.sol/TestBadDebtPushSpell.json'), output)
+copyHardhatArtifact(join(hardhatTestArtifacts, 'TestBadDebtPushSpell.sol/DaiJoinLike.json'), output)
+copyHardhatArtifact(join(hardhatTestArtifacts, 'TestBadDebtPushSpell.sol/VatLike.json'), output)
```
Comment (krzkaczor) [inline scripts/copy-artifacts.js:60]: do you need these additional artifacts?

**Review by krzkaczor — CHANGES_REQUESTED** [review]

File: test/spell.ts (hunk 2 above)
Comment (olivdb) [reply test/spell.ts:32]: Yes, good point

File: scripts/copy-artifacts.js (hunk 3 above)
Comment (olivdb) [reply scripts/copy-artifacts.js:60]: They are not needed for the tests to pass but `yarn typecheck` fails without them.

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**